Source/WebCore/platform/graphics/cg/ImageUtilitiesCG.cpp

```diff
@@ -206,82 +206,97 @@ static void tryCreateNativeImageFromData(std::span<const uint8_t> data, std::opt
+        if (!bitmap)
+            return { };
 
-        // Quality is more important than rendering speed for the current use case.
-        CGContextSetInterpolationQuality(context.get(), kCGInterpolationHigh);
-        CGContextDrawImage(context.get(), CGRectMake(0, 0, length, length), image.platformImage().get());
-        RetainPtr newImage = adoptCF(CGBitmapContextCreateImage(context.get()));
-        if (!newImage)
-            return nullptr;
-
-        CGImageDestinationAddImage(destination.get(), newImage.get(), nullptr);
+        bitmaps.append(bitmap.releaseNonNull());
     }
 
-    if (!CGImageDestinationFinalize(destination.get()))
+    return bitmaps;
+}
+
+static RefPtr<NativeImage> createNativeImageFromSVGImage(SVGImage& image, const IntSize& size)
+{
+    RefPtr buffer = ImageBuffer::create(size, RenderingMode::Unaccelerated, RenderingPurpose::Unspecified, 1, DestinationColorSpace::SRGB(), ImageBufferPixelFormat::BGRA8);
+    if (!buffer)
         return nullptr;
 
-    return SharedBuffer::create(destinationData.get());
+    Ref svgImageContainer = SVGImageForContainer::create(&image, size, 1, { });
+    buffer->context().drawImage(svgImageContainer.get(), FloatPoint::zero());
+
+    return ImageBuffer::sinkIntoNativeImage(WTFMove(buffer));
 }
 
-static RefPtr<SharedBuffer> expandSVGImageToData(SVGImage& svgImage, ASCIILiteral uti, std::span<const unsigned> lengths)
+static Vector<Ref<ShareableBitmap>> createBitmapsFromSVGImage(SVGImage& image, std::span<const unsigned> lengths)
 {
-    RetainPtr colorSpace = adoptCF(CGColorSpaceCreateWithName(kCGColorSpaceSRGB));
-    RetainPtr destinationData = adoptCF(CFDataCreateMutable(0, 0));
-    RetainPtr cfUTI = uti.createCFString();
-    RetainPtr destination = adoptCF(CGImageDestinationCreateWithData(destinationData.get(), cfUTI.get(), lengths.size(), nullptr));
+    Vector<Ref<ShareableBitmap>> bitmaps;
     for (auto length : lengths) {
-        auto size = FloatSize { float(length), float(length) };
-        RefPtr buffer = ImageBuffer::create(size, RenderingMode::Unaccelerated, RenderingPurpose::Unspecified, 1, DestinationColorSpace::SRGB(), ImageBufferPixelFormat::BGRA8);
-        if (!buffer)
-            return nullptr;
+        IntSize size { (int)length, (int)length };
+        RefPtr nativeImage = createNativeImageFromSVGImage(image, size);
+        if (!nativeImage)
+            return { };
 
-        Ref svgImageContainer = SVGImageForContainer::create(&svgImage, size, 1, { });
-        buffer->context().drawImage(svgImageContainer.get(), FloatPoint::zero());
-
-        RefPtr newImage = ImageBuffer::sinkIntoNativeImage(WTFMove(buffer));
-        if (!newImage)
-            return nullptr;
+        RefPtr bitmap = ShareableBitmap::createFromImageDraw(*nativeImage, DestinationColorSpace::SRGB(), size, ShareableBitmap::ShouldShrinkToFit::Yes);
```

Comment:
This can be written like this (See above):
``` 
RefPtr bitmap = ShareableBitmap::createFromImageDraw(*nativeImage, DestinationColorSpace::SRGB(), size, nativeImage->size());
```